core/resource/config/config.go

```diff
@@ -0,0 +1,78 @@
+package config
+
+import (
+	"fmt"
+	"maps"
+
+	"github.com/goto/shield/internal/schema"
+	"gopkg.in/yaml.v2"
+)
+
+type RoleConfig struct {
+	Name       string   `yaml:"name" json:"name"`
+	Principals []string `yaml:"principals" json:"principals"`
+}
+
+type PermissionsConfig struct {
+	Name  string   `yaml:"name" json:"name"`
+	Roles []string `yaml:"roles" json:"roles"`
+}
+
+type ResourceTypeConfig struct {
+	Name        string              `yaml:"name" json:"name"`
+	Roles       []RoleConfig        `yaml:"roles" json:"roles"`
+	Permissions []PermissionsConfig `yaml:"permissions" json:"permissions"`
+}
+
+type Config struct {
```

Comment:
any reason why this config put under `config` package and not under `resource` package?